File: TiendaOnline/TiendaOnline.Dominio/Entidades/Envio.cs
using System; // Permite utilizar tipos como DateTime.
using System.Collections.Generic; // Permite trabajar con colecciones genéricas.

namespace TiendaOnline.Dominio.Entidades; 

public partial class Envio // Representa el envío de un pedido realizado por un usuario.
{
    public int IdEnvio { get; set; } // Identificador único del envío.

    public int IdPedido { get; set; } // Guarda el identificador del pedido que se está enviando.

    public int IdDireccion { get; set; } // Guarda el identificador de la dirección donde se realizará la entrega.

    public string? EmpresaEnvio { get; set; } // Guarda el nombre de la empresa encargada del envío.

    public string? NumeroSeguimiento { get; set; } // Guarda el número utilizado para rastrear el envío.

    public DateTime? FechaEnvio { get; set; } // Guarda la fecha y hora en que se realizó el envío.

    public DateTime? FechaEntrega { get; set; } // Guarda la fecha y hora en que se entregó el pedido.

    public string Estado { get; set; } = null!; // Indica el estado actual del envío.

    public virtual DireccionUsuario IdDireccionNavigation { get; set; } = null!; // Permite acceder a la dirección relacionada con el envío.

    public virtual Pedido IdPedidoNavigation { get; set; } = null!; // Permite acceder al pedido relacionado con el envío.
}

File: TiendaOnline/TiendaOnline.Web/src/app/services/envio.ts
// Permite crear e inyectar servicios.
import {
  inject,
  Injectable
} from '@angular/core';

// Permite realizar peticiones HTTP.
import {
  HttpClient
} from '@angular/common/http';

// Permite trabajar con respuestas asíncronas.
import {
  Observable
} from 'rxjs';

// Importa el modelo de envío.
import {
  IEnvio
} from '../model/IEnvio';

// Importa la configuración.
import {
  environment
} from '../../environments/environment';

// Guarda la dirección principal.
const baseUrl =
  environment.apiUrl;

// Representa los datos de un envío.
export interface EnvioGuardar {

  // Identificador del envío.
  idEnvio: number;

  // Pedido relacionado.
  idPedido: number;

  // Dirección relacionada.
  idDireccion: number;

  // Empresa encargada.
  empresaEnvio: string | null;

  // Número de seguimiento.
  numeroSeguimiento: string | null;

  // Fecha de envío.
  fechaEnvio: string | null;

  // Fecha de entrega.
  fechaEntrega: string | null;

  // Estado actual.
  estado: string;
}

// Permite utilizar el servicio.
@Injectable({
  providedIn: 'root'
})
export class EnvioService {

  // Inyecta HttpClient.
  private readonly http =
    inject(HttpClient);

  // Obtiene el seguimiento de un pedido.
  obtenerPorPedido(
    idPedido: number
  ): Observable<IEnvio> {

    // Consulta el envío relacionado.
    return this.http.get<IEnvio>(
      `${baseUrl}/Envios/pedido/${idPedido}`
    );
  }

  // Obtiene todos los envíos.
  listar():
    Observable<IEnvio[]> {

    // Consulta los registros.
    return this.http.get<IEnvio[]>(
      `${baseUrl}/Envios`
    );
  }

  // Crea un nuevo envío.
  crear(
    envio: EnvioGuardar
  ): Observable<IEnvio> {

    // Envía el registro a la API.
    return this.http.post<IEnvio>(
      `${baseUrl}/Envios`,
      envio
    );
  }

  // Modifica un envío.
  modificar(
    envio: EnvioGuardar
  ): Observable<void> {

    // Envía los cambios.
    return this.http.put<void>(
      `${baseUrl}/Envios/${envio.idEnvio}`,
      envio
    );
  }
}

// Mantiene el nombre anterior.
export {
  EnvioService as Envio
};
